src/components/structures/AutocompleteInput.tsx

```diff
@@ -0,0 +1,219 @@
+                onMouseDown={(e) => {
+                    e.preventDefault();
+                    e.stopPropagation();
+
+                    toggleSelection(completion);
+                }}
+                key={completion.completionId}
+                data-testid={`autocomplete-suggestion-item-${completion.completionId}`}
+            >
+                <div>
+                    { children }
+                </div>
+                { isSelected && <CheckmarkIcon height={16} width={16} /> }
+            </div>
+        );
+
+        if (renderSuggestion) {
+            return withContainer(renderSuggestion(completion));
+        }
+
+        return withContainer(
+            <>
+                <span className='mx_AutocompleteInput_suggestion_title'>{ completion.completion }</span>
+                <span className='mx_AutocompleteInput_suggestion_description'>{ completion.completionId }</span>
+            </>,
+        );
+    };
+
+    const _renderSelection = (s: ICompletion): ReactNode => {
+        const withContainer = (children: ReactNode): ReactNode => (
+            <span
+                className='mx_AutocompleteInput_editor_selection'
+                key={s.completionId}
+                data-testid={`autocomplete-selection-item-${s.completionId}`}
+            >
+                <span className='mx_AutocompleteInput_editor_selection_pill'>
+                    { children }
+                </span>
+                <AccessibleButton
+                    className='mx_AutocompleteInput_editor_selection_remove'
+                    onClick={() => removeSelection(s)}
+                    data-testid={`autocomplete-selection-remove-button-${s.completionId}`}
+                >
+                    <PillRemoveIcon width={8} height={8} />
+                </AccessibleButton>
+            </span>
+        );
+
+        if (renderSelection) {
+            return withContainer(renderSelection(s));
+        }
+
+        return withContainer(
+            <span className='mx_AutocompleteInput_editor_selection_text'>{ s.completion }</span>,
+        );
+    };
+
+    const hasPlaceholder = (): boolean => selection.length === 0 && query.length === 0;
+
+    return (
+        <div className="mx_AutocompleteInput">
+            <div
+                ref={editorContainerRef}
+                className='mx_AutocompleteInput_editor'
+                onClick={onClickInputArea}
+                data-testid="autocomplete-editor"
+            >
+                { selection.map(s => _renderSelection(s)) }
+                <input
+                    ref={editorRef}
+                    type="text"
+                    onKeyDown={onKeyDown}
+                    onChange={onQueryChange}
+                    value={query}
+                    autoComplete="off"
+                    placeholder={hasPlaceholder() ? placeholder : null}
+                    onFocus={() => setFocused(true)}
+                    onBlur={() => setFocused(false)}
```

Comment:
We actually have a react hook called `useFocus` that you could use here.

Reply:
Neat, thanks for the hint 👍